### PR TITLE
add -r support to setup.py

### PR DESCRIPTION
**Pull request by cmcmarrow**
### What does this PR do?
lets salt-bin build on windows

### New Behavior
"-r <file>" is now ran in setup.py

### Tests written?
no

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by Akm0d — DISMISSED** [review]

**Review by xeacott — DISMISSED** [review]

**Review by dmurphy18 — COMMENTED** [review]
Worried that fixing this issue for 3000.1 will change current mode of operation (even if it is wrong) and now building will have a slew of new requirements appear, similar to what happened with the change from distutils to setuptools in 3000.  Perhaps this would be better postponed till Sodium.  But open to persuasion on the matter

**Review by s0undt3ch — COMMENTED** [review]
I would advise against merging this for 3000.1

There's an open PR which fixes the requirements for 3000.1 in a less intrusive way.

https://github.com/saltstack/salt/pull/56099

For Sodium, as long as the packaging is not broken, I'm ok with this change.

**Comment by cmcmarrow** [comment]
Windows and Apple are the only two that have -r from what I can see. The normal Windows installer catches all missing modules that this catches. The only place I can see go wrong is Apple and that's just because I don't know how the processes.

File: setup.py [hunk 1]
```diff
@@ -192,7 +192,12 @@ def _parse_requirements_file(requirements_file):
     with open(requirements_file) as rfh:
         for line in rfh.readlines():
             line = line.strip()
-            if not line or line.startswith(('#', '-r')):
+            if not line or line.startswith('#'):
+                continue
+            if not line or line.startswith('-r'):
```
Comment (twangboy) [inline setup.py]: Would `not line` ever be evaluated here considering the if statement above?

**Review by twangboy — CHANGES_REQUESTED** [review]

**Review by DmitryKuzmenko — DISMISSED** [review]

**Comment by s0undt3ch** [comment]
This is superseded by https://github.com/saltstack/salt/pull/56904

**Comment by DmitryKuzmenko** [comment]
@s0undt3ch maybe close it then?

**Comment by s0undt3ch** [comment]
It will get closed when the linked PR is merged.

**Comment by DmitryKuzmenko** [comment]
Hm... So this one or #56904?

**Comment by DmitryKuzmenko** [comment]
@s0undt3ch ^^^

**Comment by DmitryKuzmenko** [comment]
Just checked this part in #56904 it also handles an inline hash-comments.
I'd better like #56904

**Review by Akm0d — APPROVED** [review]

**Review by dmurphy18 — APPROVED** [review]

**Comment by s0undt3ch** [comment]
None for Sodium.
If the issue is salt-bin, we'll address it when building(again for Sodium).

**Review by DmitryKuzmenko — APPROVED** [review]

**Comment by cmcmarrow** [comment]
@s0undt3ch the PR is open due to salt-bin.

**Pull request closed without merging**